### ISSUE TITLE
[Bug]: Fooocus inpaint does not work

### ISSUE THREAD

**Issue by hben35096** [lllyasviel/stable-diffusion-webui-forge#256]
### Checklist

- [ ] The issue exists after disabling all extensions
- [X] The issue exists on a clean installation of webui
- [ ] The issue is caused by an extension, but I believe it is caused by a bug in the webui
- [X] The issue exists in the current version of the webui
- [X] The issue has not been reported before recently
- [X] The issue has been reported before but has not been fixed yet

### What happened?

Fooocus inpaint, The resulting results change very little

![PixPin_2024-02-15_01-14-59](https://github.com/lllyasviel/stable-diffusion-webui-forge/assets/139383150/8bb0d4a5-3c3a-42c7-963f-3e116e906ef1)

### Steps to reproduce the problem

![PixPin_2024-02-15_01-16-49](https://github.com/lllyasviel/stable-diffusion-webui-forge/assets/139383150/42d5f90c-1432-4def-ba92-8841713314b8)
![PixPin_2024-02-15_01-17-21](https://github.com/lllyasviel/stable-diffusion-webui-forge/assets/139383150/52ba5481-b4ad-43c3-8176-f99172d6a133)

long_hair,
Steps: 7, Sampler: DPM++ SDE Karras, CFG scale: 1.5, Seed: 2368543817, Size: 712x1024, Model hash: 7eb187c0d0, Model: dreamshaperXL_v2TurboDpmppSDE, Denoising strength: 1, Clip skip: 2, ControlNet 0: "Module: inpaint_only, Model: inpaint_v26.fooocus [be360607], Weight: 1, Resize Mode: Crop and Resize, Processor Res: 0.5, Threshold A: 0.5, Threshold B: 0.5, Guidance Start: 0, Guidance End: 0.5, Pixel Perfect: False, Control Mode: ControlNet is more important, Hr Option: Both", Mask blur: 4, Masked content: fill, Version: f0.0.12-latest-155-gd81e353d

Time taken: 11.2 sec.

### What should have happened?

![image](https://github.com/lllyasviel/stable-diffusion-webui-forge/assets/139383150/af78753b-b89a-46d9-b332-8291373f4cda)

### What browsers do you use to access the UI ?

Microsoft Edge

### Sysinfo

[sysinfo-2024-02-14-17-31.json](https://github.com/lllyasviel/stable-diffusion-webui-forge/files/14284320/sysinfo-2024-02-14-17-31.json)

### Console logs

```Shell
/root/stable-diffusion-webui-forge
Python 3.10.8 (main, Nov 24 2022, 14:13:03) [GCC 11.2.0]
Version: f0.0.12-latest-155-gd81e353d
Commit hash: d81e353d8928147bbd973068d0efbb2802affe0f
Launching Web UI with arguments: --port=6006 --theme=dark --no-half --xformers
Total VRAM 20188 MB, total RAM 257808 MB
xformers version: 0.0.23.post1
Set vram state to: NORMAL_VRAM
Device: cuda:0 NVIDIA GeForce RTX 3080 : native
VAE dtype: torch.bfloat16
/root/miniconda3/lib/python3.10/site-packages/pytorch_lightning/utilities/distributed.py:258: LightningDeprecationWarning: `pytorch_lightning.utilities.distributed.rank_zero_only` has been deprecated in v1.8.1 and will be removed in v2.0.0. You can import it from `pytorch_lightning.utilities` instead.
  rank_zero_deprecation(
Using xformers cross attention
ControlNet preprocessor location: /root/stable-diffusion-webui-forge/models/ControlNetPreprocessor
sd-webui-prompt-all-in-one background API service started successfully.
Loading weights [7eb187c0d0] from /root/stable-diffusion-webui-forge/models/Stable-diffusion/dreamshaperXL_v2TurboDpmppSDE.safetensors
2024-02-15 01:27:54,992 - ControlNet - INFO - ControlNet UI callback registered.
Running on local URL:  http://127.0.0.1:6006
model_type EPS
UNet ADM Dimension 2816
Using xformers attention in VAE
Working with z of shape (1, 4, 32, 32) = 4096 dimensions.
Using xformers attention in VAE

To create a public link, set `share=True` in `launch()`.
Startup time: 19.4s (prepare environment: 3.1s, import torch: 8.6s, other imports: 0.6s, load scripts: 3.0s, create ui: 1.2s, gradio launch: 2.4s, app_started_callback: 0.4s).
extra {'cond_stage_model.clip_l.transformer.text_model.embeddings.position_ids', 'cond_stage_model.clip_g.transformer.text_model.embeddings.position_ids', 'cond_stage_model.clip_l.logit_scale', 'cond_stage_model.clip_l.text_projection'}
left over keys: dict_keys(['conditioner.embedders.0.logit_scale', 'conditioner.embedders.0.text_projection'])
To load target model SDXLClipModel
Begin to load 1 model
Moving model(s) has taken 0.34 seconds
Model loaded in 6.4s (load weights from disk: 1.5s, forge load real models: 3.6s, load VAE: 0.2s, calculate empty prompt: 1.0s).
To load target model AutoencoderKL
Begin to load 1 model
To load target model SDXL
Begin to load 1 model
Moving model(s) has taken 0.91 seconds
  0%|                                                     | 0/7 [00:00<?, ?it/s]
 14%|██████▍                                      | 1/7 [00:00<00:04,  1.27it/s]
 29%|████████████▊                                | 2/7 [00:01<00:03,  1.66it/s]
 43%|███████████████████▎                         | 3/7 [00:01<00:02,  1.85it/s]
 57%|█████████████████████████▋                   | 4/7 [00:02<00:01,  1.95it/s]
 71%|████████████████████████████████▏            | 5/7 [00:02<00:00,  2.01it/s]
 86%|██████████████████████████████████████▌      | 6/7 [00:03<00:00,  2.06it/s]
100%|█████████████████████████████████████████████| 7/7 [00:03<00:00,  2.09it/s]

Total progress: 100%|█████████████████████████████| 7/7 [00:03<00:00,  2.03it/s]
2024-02-15 01:28:42,815 - ControlNet - INFO - ControlNet Input Mode: InputMode.SIMPLE
2024-02-15 01:28:42,884 - ControlNet - INFO - Using preprocessor: inpaint_only
2024-02-15 01:28:42,884 - ControlNet - INFO - preprocessor resolution = 0.5
Warning torch.load doesn't support weights_only on this pytorch version, loading unsafely.
2024-02-15 01:28:43,730 - ControlNet - INFO - Current ControlNet FooocusInpaintPatcher: /root/stable-diffusion-webui-forge/models/ControlNet/inpaint_v26.fooocus.patch
[Fooocus Patch Loader] 960 keys loaded, 0 remaining keys not found in model.
2024-02-15 01:28:44,518 - ControlNet - INFO - ControlNet Method inpaint_only patched.
To load target model SDXL
Begin to load 1 model
unload clone 1
Moving model(s) has taken 4.54 seconds
  0%|                                                     | 0/7 [00:00<?, ?it/s]
 14%|██████▍                                      | 1/7 [00:00<00:02,  2.13it/s]
 29%|████████████▊                                | 2/7 [00:00<00:02,  2.10it/s]
 43%|███████████████████▎                         | 3/7 [00:01<00:01,  2.10it/s]To load target model SDXL
Begin to load 1 model
unload clone 0
Moving model(s) has taken 1.52 seconds

 57%|█████████████████████████▋                   | 4/7 [00:03<00:03,  1.08s/it]
 71%|████████████████████████████████▏            | 5/7 [00:03<00:01,  1.16it/s]
 86%|██████████████████████████████████████▌      | 6/7 [00:04<00:00,  1.38it/s]
100%|█████████████████████████████████████████████| 7/7 [00:04<00:00,  1.53it/s]

Total progress: 100%|█████████████████████████████| 7/7 [00:04<00:00,  1.41it/s]
2024-02-15 01:29:08,729 - ControlNet - INFO - ControlNet Input Mode: InputMode.SIMPLE
2024-02-15 01:29:08,796 - ControlNet - INFO - Using preprocessor: inpaint_only
2024-02-15 01:29:08,796 - ControlNet - INFO - preprocessor resolution = 0.5
2024-02-15 01:29:08,963 - ControlNet - INFO - Current ControlNet FooocusInpaintPatcher: /root/stable-diffusion-webui-forge/models/ControlNet/inpaint_v26.fooocus.patch
[Fooocus Patch Loader] 960 keys loaded, 0 remaining keys not found in model.
2024-02-15 01:29:09,791 - ControlNet - INFO - ControlNet Method inpaint_only patched.
To load target model SDXL
Begin to load 1 model
unload clone 1
Moving model(s) has taken 4.08 seconds
  0%|                                                     | 0/7 [00:00<?, ?it/s]
 14%|██████▍                                      | 1/7 [00:00<00:02,  2.12it/s]
 29%|████████████▊                                | 2/7 [00:00<00:02,  2.09it/s]
 43%|███████████████████▎                         | 3/7 [00:01<00:01,  2.08it/s]To load target model SDXL
Begin to load 1 model
unload clone 0
Moving model(s) has taken 2.25 seconds

 57%|█████████████████████████▋                   | 4/7 [00:04<00:04,  1.37s/it]
 71%|████████████████████████████████▏            | 5/7 [00:04<00:02,  1.04s/it]
 86%|██████████████████████████████████████▌      | 6/7 [00:05<00:00,  1.18it/s]
100%|█████████████████████████████████████████████| 7/7 [00:05<00:00,  1.31it/s]

Total progress: 100%|█████████████████████████████| 7/7 [00:05<00:00,  1.23it/s]
```

**Comment by hben35096** [comment]
![Uploading PixPin_2024-02-15_01-39-27.png…]()

**Comment by jetjodh** [comment]
Fooocus inpaint is not supported in CN in SD WebUI

**Comment by hben35096** [comment]
> Fooocus inpaint is not supported in CN in SD WebUI
![image](https://github.com/lllyasviel/stable-diffusion-webui-forge/assets/139383150/3bc632fe-ce6e-4a5d-863a-9bfe073fb0a5)
/root/stable-diffusion-webui-forge
d81e353d8928147bbd973068d0efbb2802affe0f
I think it should be supportive

**Comment by jetjodh** [comment]
> @jetjodh It is supposed to be. [fb2e271](https://github.com/lllyasviel/stable-diffusion-webui-forge/commit/fb2e2716683453e8b0ba775f7596d8c9b080177d)

I wasn't familiar with your game.

**Comment by jetjodh** [comment]
> @jetjodh It is supposed to be. [fb2e271](https://github.com/lllyasviel/stable-diffusion-webui-forge/commit/fb2e2716683453e8b0ba775f7596d8c9b080177d)

I apologize, I was not familiar with your game.

**Comment by Maki9009** [comment]
i cant get it to produce same quality as focus idk what the correct workflow for it on forge should be

**Comment by kwas-123** [comment]
seems like it assumes a very low denoising strength. it is doing inpaint, and it is changing the image, just very very slightly

**Comment by Grufus2** [comment]
I think I might know the causation of this bug, I believe the bug is not related to Fooocus inpaint, but on its preprocessors. One small workaround I found is by doing as in Fooocus, blending the image with the mask value.

In `\webui\extensions-builtin\forge_preprocessor_inpaint\scripts\preprocessor_inpaint.py`:

For both preprocessors, in `process_before_every_sampling`:

Start of the function
```
mask = mask.round()
self.image = cond
```
to
```
mask = mask.round()
w = mask[..., None]
self.image = cond * (1.0 - w) - w
```

End of the function
```
mixed_cond = cond * (1.0 - mask) - mask
```
to
```
mixed_cond = cond * (1.0 - w) - w
```

Could someone validate this fix?
